### DISCUSSION TITLE
Question about error using interpolate_with_sva_pilotpoints_2d() provided grid specification file from modflow-usg (unstructured grid)

### DISCUSSION THREAD

**Discussion by DataGeoranger** [pypest/pypestutils#24] [Q&A] [answered]
Hello!

I am trying to use interpolate_with_sva_pilotpoints_2d() function with unstructured grid providing the grid specification file associated with modflow-usg. Please see below the screenshot of the code and error message: 
![image](https://github.com/pypest/pypestutils/assets/47303462/b2dcad7f-2070-4922-a0ea-9bd863855772)

I also tried to use the function provided by pypestutils: 'get_grid_info_from_gridspec. Then I realized that it only works for structured grid type. 

Could you please provide any suggestions on reading in the grid spec file from modflow-usg for this function or constructing [SpatialReference](https://pypest.github.io/pypestutils/helpers.html#pypestutils.helpers.SpatialReference) object in general? Any tips are much appreciated!

**Comment by hjia1005** [comment, marked as answer]
![image](https://github.com/pypest/pypestutils/assets/33449394/4534ca17-d329-4c5a-a6ff-627b616da70f)

![image](https://github.com/pypest/pypestutils/assets/33449394/537326c3-c108-4a63-8dc6-089f089b7d64)

According to the above, you don't have to provide the gsf file for an unstructured grid, a pandas dataframe with x,y, and layer will be accepted. I normally use this way for unstructured grid. Hope it helps.

**Reply by DataGeoranger** [reply]
@hjia1005 Thanks much for the tips! It works when passing x, y, layer as df.

However, the interpolate_with_sva_pilotpoints_2d() broke again, when I tried to do interpolation just for a particular K zone. 
For illustration, it works when I have pilot points covering the entire model area, and it stops working when I interpolate a sub zone. 

My guess is the correlation length for max distance for kriging probably breaks when having sub-zone pp. Another issue is related to grid_info file, there is not column for zone number for each node, I don't see get_2d_grid_info_from_file() is trying to read in zone info as a input parameter. I am not sure where to provide that, which is sort like the node_property that plproc takes, so it allows me to perform kriging just within a defined area within the model. Maybe this should be a new topic for discussion. Thanks.

![image](https://github.com/pypest/pypestutils/assets/47303462/02b0abe8-9747-49a2-a79a-539cfc81b67a)
![image](https://github.com/pypest/pypestutils/assets/47303462/fdfd8a92-2059-4f0e-a2ba-e4ffde0c9738)
![image](https://github.com/pypest/pypestutils/assets/47303462/61378fd3-a5f8-4daa-9d89-55892357065c)

**Reply by hjia1005** [reply]
@DataGeoranger In that case, I would only use the grid information (x and y) in that specific zone other than the entire model. You can do the same for multiple zones, just like what you do when using plproc in PEST utilities.

**Reply by DataGeoranger** [reply]
@hjia1005. Gotcha. So it's up to the user to provide the selected cells compared to the selection in plproc. Thanks for the tips!
